onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_matmul.cc

```diff
@@ -20,108 +26,159 @@ ONNX_OPERATOR_KERNEL_EX(
     (*KernelDefBuilder::Create())
         .TypeConstraint("Q", DataTypeImpl::GetTensorType<int8_t>())
         .TypeConstraint("S", DataTypeImpl::GetTensorType<float>())
-        .InputMemoryType(OrtMemTypeCPUInput, 1)   // scale_A
-        .InputMemoryType(OrtMemTypeCPUInput, 3)   // scale_B
-        .InputMemoryType(OrtMemTypeCPUInput, 4)   // scale_Y
-        .InputMemoryType(OrtMemTypeCPUInput, 7),  // scale_C
+        .InputMemoryType(OrtMemTypeCPUInput, QOrderedMatMulScaleA)   // scale_A
+        .InputMemoryType(OrtMemTypeCPUInput, QOrderedMatMulScaleY)   // scale_Y
+        .InputMemoryType(OrtMemTypeCPUInput, QOrderedMatMulScaleC),  // scale_C
     QOrderedMatMul);
 
-static Status ParseRowMajorTensorMetadata(const Tensor& input_tensor, int64_t& rows,
-                                          int64_t& cols, int64_t& batch_count, int64_t& element_count) {
-  const auto& dims = input_tensor.Shape().GetDims();
+QOrderedMatMul::QOrderedMatMul(const OpKernelInfo& info) : CudaKernel(info) {
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11040
+
+  order_A_ = GetCublasLtOrderAttr(info, "order_A");
```

Comment:
I think we need to add `ORT_ENFORCE(order_B_!= CUBLAST_ORDER_ROW)` seeing as the kernel does not support ROW MAJOR weight